server/routes/appointments/search/handlers/search.ts

```diff
@@ -0,0 +1,82 @@
+import { Request, Response } from 'express'
+import { Expose, Type } from 'class-transformer'
+import { IsNotEmpty, ValidateNested } from 'class-validator'
+import { isValid } from 'date-fns'
+import SimpleDate, { simpleDateFromDate } from '../../../../commonValidationTypes/simpleDate'
+import IsValidDate from '../../../../validators/isValidDate'
+import ActivitiesService from '../../../../services/activitiesService'
+import PrisonService from '../../../../services/prisonService'
+import { AppointmentOccurrenceSearchRequest } from '../../../../@types/activitiesAPI/types'
+import { toDate, toDateString } from '../../../../utils/utils'
+
+export class Search {
+  @Expose()
+  @Type(() => SimpleDate)
+  @ValidateNested()
+  @IsNotEmpty({ message: 'Enter a start date' })
+  @IsValidDate({ message: 'Enter a valid start date' })
+  startDate: SimpleDate
+
+  @Expose()
+  @Type(() => SimpleDate)
+  endDate: SimpleDate
+}
+
+export default class SearchRoutes {
+  constructor(private readonly activitiesService: ActivitiesService, private readonly prisonService: PrisonService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { startDate, endDate, timeSlot, categoryCode, locationId, prisonerNumber, createdBy, type } = req.query
+
+    if (!isValid(toDate(startDate as string))) {
```

Comment:
Are you not able to use `startDate.toRichDate()` to get a Date object here?

Reply:
Unfortunately no as it's using a query string parameter. `startDate` is a string here rather than a `SimpleDate`

Reply:
I think you might be able to apply a `@Transform` decorator to convert the string value to a `SimpleDate`. Eg:
`@Transform(({ value }) => toDate(value))`

No worries if you want to leave this for now though.